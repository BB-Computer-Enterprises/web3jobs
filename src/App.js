// import { useState, useEffect } from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";
// import { supabase } from "./lib/api";
// import Auth from "./components/Auth";
import Home from "./components/Home";
import About from './components/About';
import Companies from './components/Companies';
import PostAJob from './components/PostAJob';
import AllJobs from './components/AllJobs';
import JobPage from './components/JobPage';

function App() {
    return (
        <div className="min-w-full min-h-screen flex bg-gray-200">
            <Router>
                <nav className="flex items-center justify-between flex-wrap bg-teal-500 p-6">
                    <ul className="flex">
                        <li className="flex-1 mr-2">
                            <Link to="/">Home</Link>
                        </li>
                        <li className="flex-1 mr-2">
                            <Link to="/about">About</Link>
                        </li>
                        <li className="flex-1 mr-2">
                            <Link to="/web3-companies">Companies</Link>
                        </li>
                        <li className="flex-1 mr-2">
                            <Link to="/post-web3-job">Post A Job</Link>
                        </li>
                        <li className="flex-1 mr-2">
                            <Link to="/web3-jobs">Jobs</Link>
                        </li>
                    </ul>
                </nav>
                <Switch>
                    <Route exact path="/about">
                        <About />
                    </Route>
                    <Route exact path="/web3-companies">
                        <Companies />
                    </Route>
                    <Route exact path="/post-web3-job">
                        <PostAJob />
                    </Route>

                    {/* 
                    TODO: Implement
                    <Route>
                        <NotFound />
                    </Route> 
                    */}

                    <Route path="/web3-jobs">
                        <AllJobs />
                    </Route>
                    <Route path="/web3-jobs/:jtitle-:cname/:jid" component={JobPage} />
                    <Route path="/">
                        <Home />
                    </Route>
                </Switch>
            </Router>
        </div>
    );
}

export default App;