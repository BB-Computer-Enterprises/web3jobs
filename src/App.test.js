import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer'
import App from './App';

describe('Snapshot tests', () => {
  test('Making sure it doesnt crash', () => {
    let tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
