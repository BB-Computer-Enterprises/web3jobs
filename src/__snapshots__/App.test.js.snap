// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Making sure it doesnt crash 1`] = `
<div
  className="min-w-full min-h-screen flex items-center justify-center bg-gray-200"
>
  <div
    className="w-full h-full sm:h-auto sm:w-2/5 max-w-sm p-5 bg-white shadow flex flex-col text-base"
  >
    <span
      className="font-sans text-4xl text-center pb-2 mb-1 border-b mx-4 align-center"
    >
      Login
    </span>
    <label
      className="mt-3 mb-2 font-medium text-lg"
      htmlFor="email"
    >
      <span
        className="font-mono mr-1 text-red-400"
      >
        *
      </span>
      Email:
    </label>
    <input
      className="bg-gray-100 border py-1 px-3"
      name="email"
      required={true}
      type="email"
    />
    <label
      className="mt-3 mb-2 font-medium text-lg"
      htmlFor="password"
    >
      <span
        className="font-mono mr-1 text-red-400"
      >
        *
      </span>
      Password:
    </label>
    <input
      className="bg-gray-100 border py-1 px-3"
      name="password"
      required={true}
      type="password"
    />
    <span
      className="text-blue-600 mt-2 cursor-pointer self-end text-sm font-medium"
      onClick={[Function]}
    >
      Forgot Password?
    </span>
    <div
      className="mt-2 flex"
    >
      <span
        className="block mx-1.5 w-full rounded-md shadow-sm"
      >
        <button
          className="border w-full border-blue-600 text-blue-600 flex justify-center py-2 px-4 text-sm font-medium rounded-md hover:bg-blue-200 focus:outline-none focus:border-blue-700 focus:shadow-outline-blue active:bg-blue-700 transition duration-150 ease-in-out"
          onClick={[Function]}
          type="submit"
        >
          Sign Up
        </button>
      </span>
      <span
        className="block w-full mx-1.5 rounded-md shadow-sm"
      >
        <button
          className="flex w-full justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-blue-600 hover:bg-blue-500 focus:outline-none focus:border-blue-700 focus:shadow-outline-blue active:bg-blue-700 transition duration-150 ease-in-out"
          onClick={[Function]}
          type="button"
        >
          Sign In
        </button>
      </span>
    </div>
    <div
      className="mt-3"
    >
      <div
        className="relative"
      >
        <div
          className="absolute inset-0 flex items-center"
        >
          <div
            className="w-full mx-1.5 border-t border-gray-300"
          />
        </div>
        <div
          className="relative flex justify-center text-sm leading-5"
        >
          <span
            className="px-2 bg-white text-gray-500"
          >
            Or continue with
          </span>
        </div>
      </div>
      <div>
        <div
          className="mt-3"
        >
          <span
            className="block rounded-md shadow-sm"
          >
            <button
              className="w-3/4 mx-auto flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-blue-600 hover:bg-blue-500 focus:outline-none focus:border-blue-700 focus:shadow-outline-blue active:bg-blue-700 transition duration-150 ease-in-out"
              onClick={[Function]}
              type="button"
            >
              GitHub
            </button>
          </span>
        </div>
        <div
          className="mt-3"
        >
          <span
            className="block rounded-md shadow-sm"
          >
            <button
              className="w-3/4 mx-auto flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-blue-600 hover:bg-blue-500 focus:outline-none focus:border-blue-700 focus:shadow-outline-blue active:bg-blue-700 transition duration-150 ease-in-out"
              onClick={[Function]}
              type="button"
            >
              Google
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
